roachtest: clean up mixed version roachtest

tbg opened the pull request:
Reverts a tiny part of #27800.

Fixes #29260.
Fixes #29261.

Release note: None

tbg committed:
roachtest: don't use flags v2.0 doesn't know

Reverts a tiny part of #27800.

Fixes #29260.
Fixes #29261.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/29264)
<!-- Reviewable:end -->

tbg committed:
roachtest: disable merge queue via query that works on v2.0

We renamed the first column in 2.1.

Release note: None

tbg committed:
roachtest: run mixed version tests against v2.0.5

v2.0.0 contains many bugs that we don't need to revisit.

Release note: None

tbg committed:
roachtest: further clean up mixedVersion test

Remove spurious TODO, use a monitor instead of an errgroup, fix the TPCC
invocation which previously was barely doing anything.

tbg committed:
roachtest: remove hack from version test

It's now running against v2.0.5 which does not require the workaround.

Release note: None

tschottdorf commented:
@benesch could you take another look? More cleanup was in order.

benesch reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29264#-:-LL5a-yI1b7YJrxh7ZZ3:bnfp4nl)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

tschottdorf commented:
TestChangefeedTimestamps, fix is just being borsed

bors r=benesch
